tests/test_sshfsmount.cpp

```diff
@@ -83,46 +86,54 @@ struct SshfsMount : public mp::test::SftpServerTest
     // cooperate better, i.e., make the reader read only when a command was issued.
     auto make_exec_to_check_commands(const CommandVector& commands, std::string::size_type& remaining,
                                      CommandVector::const_iterator& next_expected_cmd, std::string& output,
-                                     bool& invoked)
+                                     bool& invoked, mp::optional<std::string>& fail_cmd, mp::optional<bool>& fail_bool)
     {
-        auto request_exec = [this, &commands, &remaining, &next_expected_cmd, &output, &invoked](ssh_channel,
-                                                                                                 const char* raw_cmd) {
+        *fail_bool = false;
+
+        auto request_exec = [this, &commands, &remaining, &next_expected_cmd, &output, &invoked, &fail_cmd,
+                             &fail_bool](ssh_channel, const char* raw_cmd) {
             invoked = false;
 
             std::string cmd{raw_cmd};
 
-            if (next_expected_cmd != commands.end())
+            if (fail_cmd && cmd.find(*fail_cmd) != std::string::npos)
+            {
+                if (fail_bool)
+                {
+                    *fail_bool = true;
+                }
+                exit_status_mock.return_exit_code(SSH_ERROR);
+            }
```

Comment:
Use `fail_invoked` rather than `fail_bool` for consistency?

Reply:
ok